To: vim_dev@googlegroups.com
Subject: Patch 7.3.257
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.257
Problem:    Not all completions are available to user commands.
Solution:   Add "color", "compiler", "file_in_path" and "locale". (Dominique
	    Pelle)
Files:	    src/ex_docmd.c, runtime/doc/map.txt


*** ../vim-7.3.256/src/ex_docmd.c	2011-07-20 15:04:52.000000000 +0200
--- src/ex_docmd.c	2011-07-20 16:10:28.000000000 +0200
***************
*** 5297,5303 ****
--- 5297,5305 ----
  {
      {EXPAND_AUGROUP, "augroup"},
      {EXPAND_BUFFERS, "buffer"},
+     {EXPAND_COLORS, "color"},
      {EXPAND_COMMANDS, "command"},
+     {EXPAND_COMPILER, "compiler"},
  #if defined(FEAT_CSCOPE)
      {EXPAND_CSCOPE, "cscope"},
  #endif
***************
*** 5310,5319 ****
--- 5312,5326 ----
      {EXPAND_EVENTS, "event"},
      {EXPAND_EXPRESSION, "expression"},
      {EXPAND_FILES, "file"},
+     {EXPAND_FILES_IN_PATH, "file_in_path"},
      {EXPAND_FILETYPE, "filetype"},
      {EXPAND_FUNCTIONS, "function"},
      {EXPAND_HELP, "help"},
      {EXPAND_HIGHLIGHT, "highlight"},
+ #if (defined(HAVE_LOCALE_H) || defined(X_LOCALE)) \
+         && (defined(FEAT_GETTEXT) || defined(FEAT_MBYTE))
+     {EXPAND_LOCALES, "locale"},
+ #endif
      {EXPAND_MAPPINGS, "mapping"},
      {EXPAND_MENUS, "menu"},
      {EXPAND_OWNSYNTAX, "syntax"},
*** ../vim-7.3.256/runtime/doc/map.txt	2011-05-19 17:25:36.000000000 +0200
--- runtime/doc/map.txt	2011-07-20 16:10:28.000000000 +0200
***************
*** 1202,1221 ****
--- 1211,1236 ----
  
  	-complete=augroup	autocmd groups
  	-complete=buffer	buffer names
+ 	-complete=color		color schemes
  	-complete=command	Ex command (and arguments)
+ 	-complete=compiler	compilers
+ 	-complete=cscope	|:cscope| suboptions
  	-complete=dir		directory names
  	-complete=environment	environment variable names
  	-complete=event		autocommand events
  	-complete=expression	Vim expression
  	-complete=file		file and directory names
+ 	-complete=file_in_path	file and directory names in |'path'|
  	-complete=filetype	filetype names |'filetype'|
  	-complete=function	function name
  	-complete=help		help subjects
  	-complete=highlight	highlight groups
+ 	-complete=locale	locale names (as output of locale -a)
  	-complete=mapping	mapping name
  	-complete=menu		menus
  	-complete=option	options
  	-complete=shellcmd	Shell command
+ 	-complete=sign		|:sign| suboptions
  	-complete=syntax	syntax file names |'syntax'|
  	-complete=tag		tags
  	-complete=tag_listfiles	tags, file names are shown when CTRL-D is hit
*** ../vim-7.3.256/src/version.c	2011-07-20 15:09:38.000000000 +0200
--- src/version.c	2011-07-20 16:30:31.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     257,
  /**/

-- 
I have to exercise early in the morning before my brain
figures out what I'm doing.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
